### ISSUE TITLE
Update configuration query syntax to have a `partName` interop added by TF

### ISSUE THREAD

**Issue by jwvanderbeck** [KSP-RO/TestFlight#125]
For many reasons, too long to discuss here, the current *shortcut* method of simply making the configuration be blank or simply a part name is bad, causes bugs in edge cases, and is a detriment to moving forward with some plans.

So as a result I want to remove that ability and instead require that a configuration always be specified and always be in the `foo = bar` style format.  TestFlight itself will add an interop value `partName` which can be used to replace the old method.

tl;dr If an existing configuration looks like `configuration = tank5m` it would change to `configuration = partName = tank5m` and if a configuration is blank it would change to explicitly call out the part name.

This **will* require changes to all configs, and as such a coordinated release with ReliasmOverhaul should be done.  That said it shouldn't be too much work to migrate (most configs already are engines and thus already use an interop setting.

Tagging @NathanKell  for awareness

**Comment by NathanKell** [comment]
I like this a lot, for the record. I would suggest perhaps adding some other automatic interops as well.

**Comment by NathanKell** [comment]
(in fact--I'm not even sure it *does* require any changes to RO, since thus far RO only does engineConfig = foo : bar )

**Comment by jwvanderbeck** [comment]
Initially addressed in  https://github.com/KSP-RO/TestFlight/commit/f9b4da138508ee4e249265f5da3b5ab9f1507e18

That adds the new interop.  I chose to use the value `kspPartName` just to be clear which "part" name we mean.  I also made it backwards compatible with blank config strings. It will now coerce a blank string into the new format when it sees it.  

Any config using a raw partName will need to be fixed though.

This issue is still open though because there is a little cleanup to be done in the query parsing.
